src/transformers/models/patchtsmixer/modeling_patchtsmixer.py

```diff
@@ -0,0 +1,2190 @@
+            input_ts = torch.reshape(
+                input_ts,
+                (batch_size, self.num_patches, self.input_size * self.patch_len),
+            )  # input_ts: [bs x num_patch x patch_len * n_vars]
+
+        patches = self.patcher(
+            input_ts
+        )  # flatten: [bs x num_patch x num_features]   common_channel/mix_channel: [bs x n_vars x num_patch x num_features]
+
+        if self.use_pe:
+            patches = patches + self.W_pos
+
+        embedding, all_hidden_states = self.mlp_mixer_encoder(patches, output_hidden_states=output_hidden_states)
+
+        return embedding, all_hidden_states
+
+
+class ForecastHead(nn.Module):
+    """Forecasting Head.
+
+    Args:
+        config (`PatchTSMixerConfig`, *mandatory*): Configuration.
+    """
+
+    def __init__(
+        self,
+        config: PatchTSMixerConfig,
+        distribution_output=None,
+    ):
+        super().__init__()
+
+        num_patches = config.num_patches
+        input_size = config.input_size
+        patch_len = config.patch_len
+        num_features = config.num_features
+        forecast_len = config.forecast_len
+        head_dropout = config.head_dropout
+        mode = config.mode
+        forecast_channel_indices = config.forecast_channel_indices
+
+        self.forecast_len = forecast_len
+        self.nvars = input_size
+        self.num_features = num_features
+        self.input_size = input_size
+        self.patch_len = patch_len
+        self.num_patches = num_patches
+        self.forecast_channel_indices = forecast_channel_indices
+
+        if self.forecast_channel_indices is not None:
+            self.forecast_channel_indices.sort()
+        self.mode = mode
+        self.distribution_output = distribution_output
+        if self.mode in ["common_channel", "mix_channel"]:
+            if distribution_output is None:
+                self.base_forecast_block = nn.Sequential(
+                    nn.Dropout(head_dropout),
+                    nn.Linear((num_patches * num_features), forecast_len),
+                )
+            else:
+                self.base_forecast_block = nn.Sequential(
+                    nn.Dropout(head_dropout),
+                    distribution_output.get_parameter_projection(num_patches * num_features),
+                )
+
+            self.flatten = nn.Flatten(start_dim=-2)
+
+        else:
+            if distribution_output is None:
+                self.base_forecast_block = nn.Sequential(
+                    nn.Dropout(head_dropout),
+                    nn.Linear((num_patches * num_features), forecast_len * input_size),
+                )
+            else:
+                self.base_forecast_block = nn.Sequential(
+                    nn.Dropout(head_dropout),
+                    distribution_output.get_parameter_projection(num_patches * num_features),
+                )
+
+            self.flatten = nn.Flatten(start_dim=1)
+
+    def forward(self, hidden_features, target=None):
```

Comment:
```suggestion
    def forward(self, hidden_features):
```
This tensors doesn't seem to be used anywhere

Reply:
https://github.com/huggingface/transformers/pull/26247/commits/9112cbf3979d1aa8528882039ffd5ebad00e1259